app/services/add_newest_time_entries.rb

```diff
@@ -0,0 +1,50 @@
+class AddNewestTimeEntries
+  def initialize(time_entries)
+    @new_time_entries = time_entries
+    @grouped_time_entries = nil
+    @employees = nil
+  end
+
+  def add_to_db
+    sort_and_group_entries
+    ensure_all_employee_are_in_db
+    add_time_entries_to_employees
+    save_information_to_db
+  end
+
+  private
+
+  def sort_and_group_entries
+    @grouped_time_entries = TimeEntriesPreparator.new(@new_time_entries).sort_and_group
+  end
+
+  def ensure_all_employee_are_in_db
+    @grouped_time_entries.map do |time_entries_per_employee|
+      employee_credentials = time_entries_per_employee[0]
```

Comment:
You can also use [`#first`](https://ruby-doc.org/core-2.4.1/Enumerable.html#method-i-first) on *Enumerable*.